conference/src/main/java/org/spine/samples/lobby/conference/ConferenceServiceImpl.java

```diff
@@ -0,0 +1,206 @@
+            final Conference publishedConference = persistedConference.toBuilder()
+                                                                      .setIsPublished(true)
+                                                                      .build();
+
+            sendConferencePublishedEvent(publishedConference);
+            final PublishConferenceResponse response = PublishConferenceResponse.newBuilder()
+                                                                                .setId(persistedConference.getId())
+                                                                                .build();
+            responseObserver.onNext(response);
+            responseObserver.onCompleted();
+
+        } else {
+
+            //TODO:2016-04-19:andrii.loboda: handle if conference is already published
+        }
+
+    }
+
+    protected BoundedContext getBoundedContext() {
+        //TODO:2016-04-14:andrii.loboda:  move to singleton and provide event bus, command bus, etc.
+        return BoundedContext.newBuilder()
+                             .build();
+    }
+
+    protected ConferenceRepository getRepository() {
+        return new ConferenceRepository();
+    }
+
+
+    private void sendConferenceCreatedEvent(Conference createdConference) {
+        final ImmutableList.Builder<Event> result = ImmutableList.builder();
+        final ConferenceCreated conferenceCreatedEvent = EventFactory.conferenceCreated(createdConference);
+        final EventContext conferenceEventContext = EventUtils.createConferenceEventContext(createdConference.getId());
+        result.add(Events.createEvent(conferenceCreatedEvent, conferenceEventContext));
+
+        sendEvents(result.build());
+    }
+
+    private void sendConferenceUpdatedEvent(Conference updatedConference) {
+        final ImmutableList.Builder<Event> result = ImmutableList.builder();
+        final ConferenceUpdated conferenceUpdatedEvent = EventFactory.conferenceUpdated(updatedConference);
+        final EventContext conferenceEventContext = EventUtils.createConferenceEventContext(updatedConference.getId());
+        result.add(Events.createEvent(conferenceUpdatedEvent, conferenceEventContext));
+//TODO:2016-04-19:andrii.loboda: looks weird
+        sendEvents(result.build());
+    }
+
+    private void sendConferencePublishedEvent(Conference publishedConference) {
+        final ImmutableList.Builder<Event> result = ImmutableList.builder();
+        final ConferencePublished conferencePublishedEvent = EventFactory.conferencePublished(publishedConference);
+        final EventContext conferenceEventContext = EventUtils.createConferenceEventContext(publishedConference.getId());
+        result.add(Events.createEvent(conferencePublishedEvent, conferenceEventContext));
+
+        sendEvents(result.build());
+    }
+
+
+    private void sendEvents(Iterable<Event> eventsToSend) {
+        for (Event event : eventsToSend) {
+            boundedContext.getEventBus()
```

Comment:
1. Extracting a variable improves readability. We don't have to read the redundant info that `eventBus` is retrieved from the BC.
2. The performance is better, too. Please see this cool [video](https://plus.google.com/u/0/+AlexanderYevsyukov/posts/6hxUNnWw352?cfem=1) made by the performance specialist.

The only drawback I see is that we have one more field, but the advantages outweigh it, aren't they? And, as we use  only an event bus, we can just replace the BC field with it for now. Of course, if we use 10 things from BC, it can be better to not have 10 more fields for them, but I don't think this is the case.